core/src/test/scala/org/apache/spark/ui/UISuite.scala

```diff
@@ -261,8 +261,10 @@ class UISuite extends SparkFunSuite {
 
       // Try a request with bad content in a parameter to make sure the security filter
       // is being added to new handlers.
+      // scalastyle:off URLConstructor
       val badRequest = new URL(
```

Comment:
To keep the existing test coverage, we need to keep `URL` because it doesn't complain while `URI` throws `URISyntaxException`.
```
[info] - add and remove handlers with custom user filter *** FAILED *** (92 milliseconds)
[info]   java.net.URISyntaxException: Illegal character in query at index 47: http://127.0.0.1:54129/test/root?bypass&invalid<=foo
```